src/node/resolver.ts

```diff
@@ -387,7 +386,7 @@ export function resolveBareModuleRequest(
           // redirect it the optimized copy.
           return resolveBareModuleRequest(root, depId, importer, resolver)
         }
-        if (!isAsset(id)) {
+        if (jsSrcRE.test(id)) {
```

Comment:
This existing `jsSrcRE` regex looked like a nice way to facilitate this logic while decoupling from the optimizer plugin. But it functions a little differently than `isAsset`, so please check it in particular. Thanks!

Reply:
The behavior should preserve, because other framework is include conditions, like svelte.

Reply:
We could inject this logic as an argument to `resolveBareModuleRequest`, but that becomes a little ugly in `src/node/server/serverPluginModuleRewrite.ts`. I tried an alternative approach via `InternalResolver` interface in 1a29a3c